### PR TITLE
server listens to specified port

### PR DESCRIPTION
**Pull request by alex-dukhno**
Done:
* create a folder for acceptance tests. It is called _acceptance_.
* create integration test when server listens to specified port

### CODE REVIEW

File: src/acceptance/java/ua/http/ws/WebServerEndToEndTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+public class WebServerEndToEndTest {
+
+    private Process webServer;
+
+    @Before
+    public void startWebServer() throws IOException, InterruptedException {
+        webServer = new ProcessBuilder("java", "-cp", "build/classes/main", "ua.http.ws.WebServer", "-p", "8080").start();
+        webServer.waitFor(5, TimeUnit.SECONDS);
+    }
```
Comment (Alex-Diez) [inline src/acceptance/java/ua/http/ws/WebServerEndToEndTest.java]: @Dmitry404, should we have use Java Process API for acceptance tests?

Comment (alex-dukhno) [inline src/acceptance/java/ua/http/ws/WebServerEndToEndTest.java:49]: @Dmitry404, should we use Java Process API for acceptance tests?

File: src/main/java/ua/http/ws/WebServer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private int port;
+    private ServerSocket serverSocket;
+
+    public WebServer(String[] params) {
+        for (int i = 0; i < params.length; i++) {
+            if ("-p".equals(params[i])) {
+                port = Integer.parseInt(params[i + 1]);
+                i += 1;
+            }
+        }
```
Comment (alex-dukhno) [inline src/main/java/ua/http/ws/WebServer.java]: @Dmitry404, should this loop be moved to ParameterParser class?

File: src/main/java/ua/http/ws/WebServer.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+public class WebServer {
+    private int port;
+    private ServerSocket serverSocket;
+
+    public WebServer(String[] params) {
+        for (int i = 0; i < params.length; i++) {
+            if ("-p".equals(params[i])) {
+                port = Integer.parseInt(params[i + 1]);
+                i += 1;
```
Comment (Dmitry404) [inline src/main/java/ua/http/ws/WebServer.java]: Didn't get why you need this line.

**Review by Dmitry404 — COMMENTED** [review]
The idea is clear. The only one thing I didn't get is a line of code. See my comment.

File: src/main/java/ua/http/ws/WebServer.java (hunk 3 above)
Comment (alex-dukhno) [reply src/main/java/ua/http/ws/WebServer.java]: It is needless in theory.
should I remove it? and just hard code 8080 to port field?

File: src/acceptance/java/ua/http/ws/WebServerEndToEndTest.java (hunk 1 above)
Comment (Dmitry404) [reply src/acceptance/java/ua/http/ws/WebServerEndToEndTest.java:49]: Why not? It's good enough so far.

File: src/main/java/ua/http/ws/WebServer.java (hunk 3 above)
Comment (Dmitry404) [reply src/main/java/ua/http/ws/WebServer.java]: Well, I would hardcode it. There is no such functionality to change the port so far. 

Leave the "p" param in ProcessBuilder, but don't read it here.

**Comment by alex-dukhno** [comment]
My current status: I am 🚫 with headers decoding
I will open new PR when develop decoding algorithm

**Comment by alex-dukhno** [comment]
@Dmitry404 any updates? 🤣

**Comment by Dmitry404** [comment]
Good try but not. Anyway, I'm glad you've eventually learnt some github tricks.

**Pull request closed without merging**